### ISSUE TITLE
False positive persists with strftime and Style/FormatStringToken

### ISSUE THREAD

**Issue by ijdickinson** [bbatsov/rubocop#5398]
I realise that issues with `strftime` and `Style/FormatStringToken` have been raised before, but I'm still seeing a false positive:

    app/models/user_selections.rb:126:23: C: Style/FormatStringToken: 
    Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).
      value.strftime('%B %Y')
                      ^^

(in case the indentation is awry, it's the `%B` token that is being highlighted).

Noting https://github.com/bbatsov/rubocop/issues/5290, I have upgraded to version 0.52.1, which I believe contains the fix for issue 5290. Just to be sure, I also tried switching my `Gemfile` to use the Github head. Behaviour was unchanged.

--------

## Expected behavior

No warning.

## Actual behavior

Warning of `Style/FormatStringToken` on the `%B` token

## Steps to reproduce the problem

I was linting this code:

    def summarise(value)
      if value.is_a?(Date)
        value.strftime('%B %Y')
      elsif value.is_a?(Array)
        value.map(&method(:format_summary_value)).join(' ')
      else
        value.to_s
      end
    end

## RuboCop version

```
$ rubocop -V
0.52.1 (using Parser 2.4.0.2, running on ruby 2.4.2 x86_64-linux)
```

**Comment by AlexWayfer** [comment]
Also for `%d` and `%e` tokens.

Examples of strings: `'%Y-%m-%d'`, `'%e of %B'`.

**Comment by aks** [comment]
```lib/migrations/misc_support_utils.rb:18:42: C: Style/FormatStringToken: Prefer template tokens (like %{foo}) over unannotated tokens (like %s).
  6     $sql_log.puts DateTime.now.strftime("%c")
```
I'm confused why `StyleFormatStringToken` logic is being applied to `strftime` tokens  at all— they are *not* the same as `printf` tokens.

**Comment by Drenmi** [comment]
This seems fixed in `master`.
